### PR TITLE
Use Bootstrap modal for new aspect form

### LINKED ISSUE #4939: Add new aspect box does not close on aspect creation
When creating a new aspect via the aspect dropdown -> Add new aspect, after a successful create it will not close the dropdown or update the page with changed data (new aspect + number of aspects person is in).

Actual aspect creation works as per screenshot returned 200 status from call. The dropdown input is cleared.

pic

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1811793-add-new-aspect-box-does-not-close-on-aspect-creation?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by svbergerem**
Fixes #4939.

### CODE REVIEW

File: app/assets/javascripts/app/views/aspect_membership_view.js [hunk 1]
```diff
@@ -13,12 +13,20 @@ app.views.AspectMembership = app.views.AspectsDropdown.extend({
...
     "click ul.aspect_membership.dropdown-menu > li.aspect_selector": "_clickHandler",
-    "keypress ul.aspect_membership.dropdown-menu > li.aspect_selector": "_clickHandler"
+    "keypress ul.aspect_membership.dropdown-menu > li.aspect_selector": "_clickHandler",
+    "click ul.aspect_membership.dropdown-menu > li.newItem": "showModal"
   },
 
   initialize: function() {
     this.list_item = null;
     this.dropdown  = null;
+    if(this.$(".newAspectContainer").length > 0){
```
Comment (jhass) [inline app/assets/javascripts/app/views/aspect_membership_view.js]: Mh, JSHint doesn't enforce consistent spacing inside if expressions? Personally I prefer `if (condition) {`

Comment (svbergerem) [inline app/assets/javascripts/app/views/aspect_membership_view.js]: No, I guess we would have to use jscs to test that. However, I agree that `if (condition) {` would be better.

**Comment by svbergerem** [comment]
Ready to review.

**Comment by jhass** [comment]
Thanks1

**Pull request merged**